api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+
+      schema:
+        type: object
+        $ref: '#/definitions/EventTypeSchema'
+        description: |
+          The schema for this EventType. This is expected to be a json schema in yaml format (other formats
+          might be added in the future).
+
+      data_key:
+        type: array
+        items:
+          type: string
+        description: |
+          Indicators of the path of the properties that constitute the primary key (identifier) of the data
+          within this Event.
+          If set MUST be a valid required field as defined in the schema.
+          (TBD should be required? Is applicable only to both Business and DataChange Events?)
+
+      ordering_key:
+        type: string
+        description: |
+          Indicator of the field used for guaranteeing the ordering of Events of this type (used by the
+          PartitionResolutionStrategy).
+          If set MUST be a valid required field as defined in the schema.
+
+    required:
+      - name
+
+  EventTypeSchema:
+    properties:
+      type:
+        type: string
+        example: 'json-schema'
+        description: The type of schema definition (avro, json-schema, etc).
+      schema:
+        type: string
+        description: |
+          The schema as string in the syntax defined in the field type. Failure to respect the syntax will
+          fail any operation on an EventType.
+    required:
+      - type
+      - schema
+
+  EventValidationStrategy:
+    description: |
+      Defines a rule for validation of an incoming Event.
+
+      Rules might require additional parameters; see the `doc` field of the existing rules for details. See
+      GET /registry/validation-strategies for a list of available rules.
+    properties:
+      name:
+        type: string
+        description: Name of the strategy.
+      doc:
+        type: string
+        description: Documentation for the validation.
+    required:
+      - name
+
+
+  PartitionResolutionStrategy:
+    description: |
+      Defines a rule for the resolution of incoming Events into partitions of a Topic.
+
+      Rules might require additional parameters; see the `doc` field of the existing rules for details. See
+      GET /registry/partition-strategies for a list of available rules.
+    properties:
+      name:
+        type: string
+        description: Name of the strategy.
+      doc:
+        type: string
+        description: Documentation for the partition resolution.
+    required:
+      - name
+
+  EventEnrichmentStrategy:
+    description: |
+      Defines a rule for transformation of an incoming Event. No existing fields might be modified. In
+      practice this is used to set automatically values in the Event upon reception (i.e. set a reception
+      timestamp on the Event).
+
+      Rules might require additional parameters; see the `doc` field of the existing rules for details. See
+      GET /registry/enrichment-strategies for a list of available rules.
```

Comment:
Without the notion of enrichment we can't provide some of the guarantees in metadata that Saiki expects. This is more or less the "technical means" with which we provide this.